inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_imad.cpp

```diff
@@ -185,15 +188,35 @@ bool ConvolutionKernel_imad::Validate(const Params& params, const optional_param
         return false;
     }
 
-    auto& newParams = static_cast<const convolution_params&>(params);
-    if (newParams.groups > 1 && newParams.weights.IFM().v % 4 != 0 &&
-        newParams.inputs[0].GetLayout() != DataLayout::b_fs_yx_fsv16)
+    auto& conv_params = static_cast<const convolution_params&>(params);
+    if (conv_params.groups > 1 && conv_params.weights.IFM().v % 4 != 0 &&
+        conv_params.inputs[0].GetLayout() != DataLayout::b_fs_yx_fsv16)
         return false;
 
-    size_t min_block_size_x = (newParams.weights.X().v - 1) * newParams.dilation.x + 1;
+    size_t min_block_size_x = (conv_params.weights.X().v - 1) * conv_params.dilation.x + 1;
     if (min_block_size_x > SIMD_SIZE)
         return false;
 
+    if (conv_params.quantization == QuantizationType::ASYMMETRIC_DATA_AND_WEIGHTS) {
+        if ((conv_params.activations_zero_points.empty() || conv_params.weights_zero_points.empty()) &&
+            (conv_params.compensation.empty()))
+            return false;
+    }
+    else if (conv_params.quantization == QuantizationType::ASYMMETRIC_DATA) {
```

Comment:
nit: please move `else if` to the previous line:
`    } else if (conv_params.quantization == QuantizationType::ASYMMETRIC_DATA) {`